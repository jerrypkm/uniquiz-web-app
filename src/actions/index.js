export const addToState = (store, data) => {
  const state = {
    ...store.state,
    ...data,
  };
  store.setState(state);
};
export const addToCollections = (store, data) => {
  const collections = {
    ...store.state.collections,
    ...data,
  };
  store.setState({ collections });
};

export const addToAuthUser = (store, data) => {
  let authUser = {
    ...store.state.authUser,
    ...data,
  };

  if (!data) {
    authUser = null;
  }
  store.setState({ authUser });
};

export const addToUserRole = (store, data) => {
  let userRole = {
    ...store.state.userRole,
    ...data,
  };

  if (!data) {
    userRole = null;
  }
  store.setState({ userRole });
};

export const addToFirebase = (store, data) => {
  const firebaseCache = {
    ...store.state.firebaseCache,
    ...data,
  };
  store.setState({ firebaseCache });
};

export const addToCourses = (store, data) => {
  const courses = {
    ...store.state.courses,
    ...data,
  };
  store.setState({ courses });
};

export const addToThemeOptions = (store, data) => {
  const themeOptions = {
    ...store.state.themeOptions,
    ...data,
  };
  store.setState({ themeOptions });
};
