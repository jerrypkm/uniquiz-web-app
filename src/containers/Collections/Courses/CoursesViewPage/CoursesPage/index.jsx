/**
 * Collections is a container that fetches firebase data using hooks and renders a list of all collections
 * @param {Object} authUser - Passed from parent container and has everything about the logged in user
 * @param {Object} collection - Passed from parent with a title and a path of the collection
 * @returns {<CoursePage/>} - returns CoursePage component which contains the rest of the components
 */

import React, { lazy } from "react";

import Footer from "components/shared/Footer";
import MoPage from "components/library/MoPage";

import Container from "@material-ui/core/Container";
import Grid from "@material-ui/core/Grid";
import Navigation from "components/shared/Navigation";
import NewItemCard from "components/shared/CardList/NewItemCard";

const CardList = lazy(() => import("components/shared/CardList"));

const CoursesPage = ({
  authUser,
  collectionDetails,
  courses,
  firebase,
  isLoading,
  isAdmin,
  itemOptions
}) => {
  if (isLoading) {
    return null;
  }

  return (
    <Container maxWidth="lg">
      <Navigation authUser={authUser} firebase={firebase} />
      <MoPage title={collectionDetails?.title}>
        <div> ptmmm</div>
        <Grid container spacing={4} alignItems="center">
          <CardList
            isAdmin={isAdmin}
            items={courses}
            itemOptions={itemOptions}
          />
          {isAdmin && (
            <Grid item xs={12} sm={12} md={4} lg={4}>
              <NewItemCard
                type="new"
                title={itemOptions?.newItem?.title}
                url={itemOptions?.newItem?.url}
              />
            </Grid>
          )}
        </Grid>
        <Footer />
      </MoPage>
    </Container>
  );
};

export default CoursesPage;
