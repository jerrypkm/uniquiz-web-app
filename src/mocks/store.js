// Mock functionality of global and store hooks

export const storeMock = jest.mock("store", () =>
  jest.fn().mockReturnValue([{ state: { authUser: null } }])
);
export const globalHookMock = jest.mock("use-global-hook", () => jest.fn([]));
