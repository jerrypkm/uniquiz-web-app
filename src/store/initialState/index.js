import { getCache } from "helpers/localCache";

const isDarkMode = getCache({ key: "isDarkMode" }) || false;

export const initialState = {
  authUser: null,
  collections: {},
  courses: [],
  themeOptions: {
    isDarkMode,
  },
  firebase: null,
};
