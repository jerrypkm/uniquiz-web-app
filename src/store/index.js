import React from "react";

import * as actions from "actions";
import { initialState } from "./initialState";
import useGlobalHook from "use-global-hook";

const useGlobal = useGlobalHook(React, initialState, actions);

export default useGlobal;
