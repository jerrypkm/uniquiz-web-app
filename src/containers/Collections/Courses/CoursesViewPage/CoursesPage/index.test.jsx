import React from "react";
import CoursesPage from "./index";

// Dependencies under test
import ShallowRenderer from "react-test-renderer/shallow";
import useGlobal from "store";
import useGlobalHook from "use-global-hook";

import { renderHook } from "@testing-library/react-hooks";

import { initialState } from "store/initialState";
import actions from "actions";

// Shallow render to test the component without it's children
const renderer = new ShallowRenderer();

beforeEach(() => {
  useGlobal.mockClear();
  useGlobalHook.mockClear();
});

afterEach(() => {
  jest.restoreAllMocks();
});

// Mock functionality of global and store hooks
jest.mock("store", () => jest.fn().mockReturnValue([]));
jest.mock("use-global-hook", () => jest.fn());

describe("CoursesPage Component", () => {
  test("should match the snapshot", () => {
    renderer.render(<CoursesPage />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  test("Should import and invoke useGlobal", () => {
    // Call useGlobal hook
    const { result, waitForNextUpdate } = renderHook(() =>
      useGlobal(React, initialState, actions)
    );
    // Wait for renders
    waitForNextUpdate();
    expect(result.current).toBeDefined();
    expect(useGlobal).toHaveBeenCalledWith(React, initialState, actions);
  });
});
