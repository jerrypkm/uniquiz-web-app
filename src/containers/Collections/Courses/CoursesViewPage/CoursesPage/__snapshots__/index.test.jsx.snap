// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoursesPage Component should match the snapshot 1`] = `
<ForwardRef(WithStyles)
  maxWidth="lg"
>
  <ForwardRef(WithStyles) />
  <ForwardRef(WithStyles)>
    <ForwardRef(WithStyles)
      alignItems="center"
      container={true}
      spacing={4}
    >
      <UNDEFINED />
    </ForwardRef(WithStyles)>
    <ForwardRef(WithStyles) />
  </ForwardRef(WithStyles)>
</ForwardRef(WithStyles)>
`;
